File: Assembly-FSharp/Assembly-FSharp/Commands/InputAxisCommand.fs
namespace Styx.Commands

open UnityEngine

type InputAxisCommand() =
    inherit Command ()

    [<SerializeField>]
    let mutable inputAxis: string = ""
    [<SerializeField>]
    let mutable value: float32 = 0.f

    member this.Value = value

    override this.Execute () =
        value <- Input.GetAxis inputAxis

    

File: Assets/_Scripts/Commands/InputAxisCommand.cs
using UnityEngine;

namespace Styx.Commands
{
    public class InputAxisCommand : Command
    {
        [SerializeField]
        private string _inputAxis = "";
        [SerializeField]
        private float _value = 0f;

        public float Value
        {
            get { return _value; }
            set { _value = value; }
        }

        public override void Execute()
        {
            _value = Input.GetAxis(_inputAxis);
        }
    }
}
